common/models/user.js

```diff
@@ -301,6 +301,21 @@ module.exports = function(User) {
     return fn.promise;
   };
 
+  User.observe('before delete', function(ctx, next) {
+    var AccessToken = ctx.Model.relations.accessTokens.modelTo;
+    ctx.Model.find({ where: ctx.where }, function(err, list) {
+      if (err) return next(err);
+
+      var pkName = ctx.Model.definition.idName() || 'id';
+      var ids = list.map(function(u) { return u[pkName]; });
+      ctx.where = {};
+      ctx.where[pkName] = { inq: ids };
+
+      AccessToken.destroyAll({ userId: { inq: ids }}, next);
```

Comment:
I previously used this code, but Miroslav had his concerns about it in this comment: https://github.com/strongloop/loopback/pull/2492#discussion_r71323039
```
  User.observe('before delete', function(ctx, next) {
    var AccessToken = ctx.Model.relations.accessTokens.modelTo;

    AccessToken.find({
      where: {
        userId: ctx.where.id,
      },
    }, function(err, tokens) {
      tokens.forEach(function(token) {
        AccessToken.destroyById(token.id, function() {
          console.log('Deleted token %s for user ', token.id, token.userId);
        });
      });
    });

    next();
  });
```